controllers/replication/replication.go

```diff
@@ -120,3 +120,15 @@ func (r *ReplicationResponse) HasKnownGRPCError(knownErrors []codes.Code) bool {
 
 	return false
 }
+
+// getMessageFromError returns the message from the error.
+func GetMessageFromError(err error) string {
+	s, ok := status.FromError(err)
```

Comment:
Do we want to check if the error is nil?

Reply:
internally it's handled. if err is nil empty string will be returned.